nvd/enricher.go

```diff
@@ -21,11 +21,41 @@ const (
 
 // FillMissingData fills NVD CVEs with missing data, e.g. affected product versions
 func FillMissingData(project Project, cves []*schema.NVDCVEFeedJSON10DefCVEItem) []*schema.NVDCVEFeedJSON10DefCVEItem {
-	cves = addMissingCVEs(project, cves)
+	cves = filterCVEs(project, addMissingCVEs(project, cves))
 	tryUpdateCVEWithAffectedVersions(cves)
+	if project == Istio {
+		// Only do this for Istio to ensure other vulns are not affected at this time.
+		populateCVSSv2IfMissing(cves)
+	}
 	return cves
 }
 
+func filterCVEs(project Project, cves []*schema.NVDCVEFeedJSON10DefCVEItem) []*schema.NVDCVEFeedJSON10DefCVEItem {
+	filtered := cves[:0]
+	for _, cve := range cves {
+		if cve == nil {
+			log.Warnf("Filtered out nil %s CVE", project)
+			continue
+		}
+		// Note: it's ok if PublishedDate and LastModifiedDate are nil.
+		if cve.CVE == nil || cve.Impact == nil || cve.Configurations == nil {
+			log.Warnf("Filtered out %s CVE due to missing field: %v", project, *cve)
+			continue
+		}
+		if cve.CVE.CVEDataMeta == nil || cve.CVE.CVEDataMeta.ID == "" {
+			log.Warnf("Filtered out %s CVE due to missing or invalid CVEDataMeta: %v", project, *cve)
+			continue
+		}
+		// Note: We already assumed there are both CVSSv2 and CVSSv3 scores for Istio, so let's just enforce there is CVSSv3.
+		if project == Istio && cve.Impact.BaseMetricV3 == nil {
+			log.Warnf("Filtered out %s CVE %s due to missing CVSSv3 score", project, cve.CVE.CVEDataMeta.ID)
```

Comment:
More of a product strategy question. If we know that cvss v3/3.1 are the only adopted versions due to the shortcomings of v2, should we drop support for cvss v2 altogether? 
(Seems like we are also lagging behind by a version for our scans)